smart_open/utils.py

```diff
@@ -94,13 +94,13 @@ def clamp(value, minval, maxval):
     return max(min(value, maxval), minval)
 
 
-def make_range_string(start, stop=None):
+def make_range_string(start=None, stop=None):
     """Create a byte range specifier in accordance with RFC-2616.
 
     Parameters
     ----------
-    start: int
-        The start of the byte range
+    start: int, optional
+        The start of the byte range.  If unspecified, stop indicated offset from EOF.
```

Comment:
Can you point to the part of the [RFC](https://tools.ietf.org/html/rfc2616) that describes this?

I couldn't find anything about this use case (start is None, stop is non-None).

```
   If the last-byte-pos value is present, it MUST be greater than or
   equal to the first-byte-pos in that byte-range-spec, or the byte-
   range-spec is syntactically invalid. The recipient of a byte-range-
   set that includes one or more syntactically invalid byte-range-spec
   values MUST ignore the header field that includes that byte-range-
   set.
```

Reply:
Sure, it's on the next page:

```
       suffix-byte-range-spec = "-" suffix-length
       suffix-length = 1*DIGIT

   A suffix-byte-range-spec is used to specify the suffix of the
   entity-body, of a length given by the suffix-length value. (That is,
   this form specifies the last N bytes of an entity-body.) If the
   entity is shorter than the specified suffix-length, the entire
   entity-body is used.
```

FWIW this behavior is confirmed by `test_seek_end`.

Reply:
I added a `seek(0)` call to __init__, and a `defer_seek` parameter that allows that call to be skipped.